pkg/protocols/utils/variables.go

```diff
@@ -172,3 +176,10 @@ func generateVariables(inputURL *urlutil.URL, removeTrailingSlash bool) map[stri
 	}
 	return generators.MergeMaps(knownVariables, GenerateDNSVariables(parsed.Hostname()))
 }
+
+func endpoint(hostname, port string) string {
+	if port != "" {
+		return fmt.Sprintf("%s:%s", hostname, port)
```

Comment:
I think https://pkg.go.dev/net#JoinHostPort is preferred as it escapes correctly ipv6s